### PR TITLE
pay: fix corner case where we eliminate our own hint.

### PR DESCRIPTION
**Pull request by rustyrussell**
Reported-by: @sgeisler

### CODE REVIEW

File: tests/test_pay.py [hunk 1]
```diff
@@ -4237,3 +4237,26 @@ def test_unreachable_routehint(node_factory, bitcoind):
     # both directly, and via the routehints we should now just have a
     # single attempt.
     assert(len(excinfo.value.error['attempts']) == 1)
+
+
+def test_routehint_tous(node_factory, bitcoind):
+    """
+Test bug where trying to pay an invoice from an *offline* node which
+gives a routehint straight to us causes an issue
+"""
```
Comment (cdecker) [inline tests/test_pay.py:4246]: Nit: indentation

**Review by cdecker — COMMENTED** [review]
ACK https://github.com/ElementsProject/lightning/pull/4555/commits/4cee8eae3edc48730689811c1c9581f6ca1c0148

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/c2b355edadd7e3d3fd04bf20e23f1760615ca87f

(@cdecker's trivial rebase)
